pkg/provider/vault/vault.go

```diff
@@ -899,7 +905,7 @@ func (v *client) readSecret(ctx context.Context, path, version string) (map[stri
 			return nil, errors.New(errDataField)
 		}
 		if dataInt == nil {
-			return nil, nil
+			return nil, esv1beta1.NoSecretError{}
```

Comment:
A bug in Vault Provider with NoSecretError is also fixed here and on `dataFrom.Find` methods